checks.d/spark.py

```diff
@@ -306,6 +422,40 @@ def _spark_stage_metrics(self, running_apps, addl_tags):
                 self._set_metrics_from_json(tags, stage, SPARK_STAGE_METRICS)
                 self._set_metric('spark.stage.count', INCREMENT, 1, tags)
 
+                stage_id = stage['stageId']
+                previous_status = None
+                if app_id in self.previous_stages and stage_id in self.previous_stages[app_id]:
+                    previous_status = self.previous_stages[app_id][stage_id]['status']
+                self._event_for_stage_status_change(stage, tags, previous_status)
+
+            # build index by mapping app ids to a mapping of job id => jobs
+            new_stages[app_id] = dict((stage['stageId'], stage) for stage in response)
+
+        self.previous_stages = new_stages
+
+    def _event_for_stage_status_change(self, current_stage, tags, previous_status):
```

Comment:
Could we maybe refactor _event_for_stage_status_change() and _event_for_job_status_change() into a single PR? A lot of the code is duplicated, being very very similar functions. It could streamline the source a bit.